### PR TITLE
JSON base controller

### PR DESCRIPTION
**Pull request by stefanorosanelli**
This PR introduces a simple abstract base class to be extended by controllers using plain JSON and not JSON-API as input or output format.

### CODE REVIEW

File: plugins/BEdita/API/src/Controller/JsonBaseController.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    {
+        parent::initialize();
+        if ($this->components()->has('JsonApi')) {
+            $this->components()->unload('JsonApi');
+        }
+        $this->viewBuilder()->setClassName('Json');
+        $this->RequestHandler->setConfig('viewClassMap.json', 'Json');
+
+        if ($this->request->contentType() === 'application/json') {
+            $this->RequestHandler->setConfig('inputTypeMap.json', ['json_decode', true], false);
```
Comment (batopa) [inline plugins/BEdita/API/src/Controller/JsonBaseController.php]: Why don't  we always set this config?

File: plugins/BEdita/API/tests/TestCase/Controller/JsonBaseControllerTest.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
+<?php
+namespace BEdita\API\Test\TestCase\Controller;
+
```
Comment (batopa) [inline plugins/BEdita/API/tests/TestCase/Controller/JsonBaseControllerTest.php:14]: [minor] missing disclaimer

File: plugins/BEdita/API/tests/TestCase/Controller/JsonBaseControllerTest.php [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+use BEdita\API\Controller\JsonBaseController;
+use Cake\Http\ServerRequest;
+use Cake\TestSuite\TestCase;
+
+/**
+ * BEdita\API\Controller\JsonBaseController Test Case
+ *
+ * @uses \BEdita\API\Controller\JsonBaseController
```
Comment (batopa) [inline plugins/BEdita/API/tests/TestCase/Controller/JsonBaseControllerTest.php]: Maybe `@coversDefaultClass`?

**Review by batopa — CHANGES_REQUESTED** [review]

File: plugins/BEdita/API/src/Controller/JsonBaseController.php (hunk 1 above)
Comment (stefanorosanelli) [reply plugins/BEdita/API/src/Controller/JsonBaseController.php]: In `AppController` we have 

```php
        if ($this->request->is(['json', 'jsonapi'])) {
            ....
            $this->RequestHandler->setConfig('inputTypeMap.json', [[$this->JsonApi, 'parseInput']], false);
            ....
       }
```
But then we unload `JsonApi` component - I thought that a `RequestHandler` config update is safer

File: plugins/BEdita/API/tests/TestCase/Controller/JsonBaseControllerTest.php (hunk 2 above)
Comment (stefanorosanelli) [reply plugins/BEdita/API/tests/TestCase/Controller/JsonBaseControllerTest.php:14]: done

File: plugins/BEdita/API/tests/TestCase/Controller/JsonBaseControllerTest.php (hunk 3 above)
Comment (stefanorosanelli) [reply plugins/BEdita/API/tests/TestCase/Controller/JsonBaseControllerTest.php]: done

**Review by batopa — APPROVED** [review]

**Pull request merged**